### PR TITLE
Getting EE data with only using plugin

### PR DESCRIPTION
**Pull request by adeelq**
Linked with https://github.com/dotmailer/dotmailer-magento2-extension/pull/480

### CODE REVIEW

**Comment by cdiacon** [comment]
1. there is no check for data mapping before setting the EE values, for ex. no need to set the "reward_ammount" if this values was not mapped in data mapping.
2. no use of magic methods $customer->setData('reward_points') -> $customer->setRewardPoints()
3. plugin in di.xml is disabled by default no need to set "disabled="false"
4. move the helper isEnterprise to EE as it checks only when need to render on EE editions.

File: Helper/Config.php [hunk 1]
```diff
@@ -0,0 +1,19 @@
...
+    const XML_PATH_CONNECTOR_ENTERPRISE_REWARD_AMOUNT = 'connector_data_mapping/enterprise_data/reward_ammount';
+    const XML_PATH_CONNECTOR_ENTERPRISE_EXPIRATION_DATE = 'connector_data_mapping/enterprise_data/expiration_date';
+    const XML_PATH_CONNECTOR_ENTERPRISE_LAST_USED_DATE = 'connector_data_mapping/enterprise_data/last_used_date';
+    const XML_PATH_CONNECTOR_ENTERPRISE_CUSTOMER_SEGMENTS = 'connector_data_mapping/enterprise_data/customer_segments';
+
+    public function __construct(Context $context)
+    {
+        parent::__construct($context);
+    }
+}
```
Comment (cdiacon) [inline Helper/Config.php]: @adeelq will this work 

```
<?php

namespace Dotdigitalgroup\Enterprise\Helper;


class Config
{
    const XML_PATH_CONNECTOR_ENTERPRISE_CURRENT_BALANCE = 'connector_data_mapping/enterprise_data/reward_points';
    const XML_PATH_CONNECTOR_ENTERPRISE_REWARD_AMOUNT = 'connector_data_mapping/enterprise_data/reward_ammount';
    const XML_PATH_CONNECTOR_ENTERPRISE_EXPIRATION_DATE = 'connector_data_mapping/enterprise_data/expiration_date';
    const XML_PATH_CONNECTOR_ENTERPRISE_LAST_USED_DATE = 'connector_data_mapping/enterprise_data/last_used_date';
    const XML_PATH_CONNECTOR_ENTERPRISE_CUSTOMER_SEGMENTS = 'connector_data_mapping/enterprise_data/customer_segments';

}
```

File: Helper/Data.php [hunk 2]
```diff
@@ -0,0 +1,151 @@
+<?php
+
+namespace Dotdigitalgroup\Enterprise\Helper;
+
+use Magento\Framework\App\Helper\Context;
+use Dotdigitalgroup\Enterprise\Helper\Config;
```
Comment (cdiacon) [inline Helper/Data.php]: @adeelq I think you can remove this line :

use Dotdigitalgroup\Enterprise\Helper\Config;

File: Plugin/DataFieldPlugin.php [hunk 3]
```diff
@@ -0,0 +1,37 @@
...
+ */
+class DataFieldPlugin
+{
+    /**
+     * @var \Dotdigitalgroup\Enterprise\Helper\Data
+     */
+    private $helper;
+
+    /**
+     * DdgHelperPlugin constructor.
```
Comment (cdiacon) [inline Plugin/DataFieldPlugin.php]: @adeelq rename this ^^^ DdgHelperPlugin as it doesn't exists anymore

File: Plugin/HelperPlugin.php [hunk 4]
```diff
@@ -0,0 +1,49 @@
...
+ */
+class HelperPlugin
+{
+    /**
+     * @var \Dotdigitalgroup\Enterprise\Helper\Data
+     */
+    private $helper;
+
+    /**
+     * DdgHelperPlugin constructor.
```
Comment (cdiacon) [inline Plugin/HelperPlugin.php]: rename this ^^^ DdgHelperPlugin as it doesn't exists anymore

File: Plugin/CustomerPlugin.php [hunk 5]
```diff
@@ -0,0 +1,229 @@
...
+     */
+    private $customer;
+
+    /**
+     * @var \Dotdigitalgroup\Enterprise\Helper\Data
+     */
+    private $helper;
+
+    /**
+     * DdgCustomerPlugin constructor.
```
Comment (cdiacon) [inline Plugin/CustomerPlugin.php]: rename this ^^^ DdgCustomerPlugin as it doesn't exists anymore

**Comment by cdiacon** [comment]
@adeelq still there same data for all contacts when you map the EE datafieds : 

* expiration date
* reward amount
* reward points

this data fields contain the values from one of the contact/customer which has this data values!!!

**Comment by adeelq** [comment]
@cdiacon ok i got it. this is fixed in other or but nit in this one. change coming son

**Comment by adeelq** [comment]
@cdiacon all changes in

File: Plugin/CustomerPlugin.php [hunk 6]
```diff
@@ -0,0 +1,230 @@
...
+
+        return '';
+    }
+
+    /**
+     * Get the customer reward.
+     *
+     * @return void
+     */
+    public function _setReward()
```
Comment (cdiacon) [inline Plugin/CustomerPlugin.php]: @adeelq this is a public function why using _underscore _setReward ?

Comment (adeelq) [reply Plugin/CustomerPlugin.php]: @cdiacon not a new code. just copied it.

Comment (cdiacon) [reply Plugin/CustomerPlugin.php]: remove underscore

**Comment by adeelq** [comment]
@cdiacon DONE

File: etc/di.xml [hunk 7]
```diff
@@ -0,0 +1,17 @@
...
+<config xmlns:xsi="http://www.w3.org/2001/XMLSchema-instance" xsi:noNamespaceSchemaLocation="urn:magento:framework:ObjectManager/etc/config.xsd">
+    <type name="Dotdigitalgroup\Email\Model\Apiconnector\Customer">
+        <plugin name="ddg_enterprise_plugin" type="Dotdigitalgroup\Enterprise\Plugin\CustomerPlugin"
+                sortOrder="1"/>
+    </type>
+    <type name="Dotdigitalgroup\Email\Helper\Data">
+        <plugin name="ddg_helper_plugin" type="Dotdigitalgroup\Enterprise\Plugin\HelperPlugin" sortOrder="1"/>
+    </type>
+    <type name="Dotdigitalgroup\Email\Model\Connector\Datafield">
+        <plugin name="ddg_data_field_plugin" type="Dotdigitalgroup\Enterprise\Plugin\DataFieldPlugin" sortOrder="1"/>
```
Comment (cdiacon) [inline etc/di.xml]: @adeelq so 3 plugins name that doesn't follow any convention ? : 
can we change to  : 
1. ddg_enterprise_plugin line 4
2. ddg_enterprise_helper_plugin line 8
3. ddg_enterprise_data_field_plugin line 11

**Pull request merged**